trackers/node-tracker/test/tracker.ts

```diff
@@ -604,4 +608,64 @@ for (const method of testMethods) {
       );
     });
   });
+
+  test(method + ' method: setSessionId should attach a sid property to event', async (t) => {
+    const expected = {
+      sid: 'sid-test-1234',
+    };
+
+    await new Promise((resolve, reject) => {
+      const e = gotEmitter(
+        endpoint,
+        HttpProtocol.HTTPS,
+        undefined,
+        method,
+        0,
+        undefined,
+        undefined,
+        function (error, response) {
+          checkPayload(extractPayload(response?.body, method), expected, t);
+          if (error) reject(error);
+          else resolve(response);
+        }
+      );
+
+      const track = tracker(e, 'cf', 'cfe35', false);
+      track.setSessionId(expected.sid);
+      track.track(
+        buildPageView({ pageUrl: 'http://www.example.com', pageTitle: 'example page', referrer: 'http://google.com' }),
+        context
+      );
+    });
+  });
+
+  test(method + ' method: setSessionIndex should attach a vid property to event', async (t) => {
+    const expected = {
+      vid: '1234',
+    };
+
+    await new Promise((resolve, reject) => {
```

Comment:
I will keep the improvement of making this a `createTracker` function for a next PR